src/cluster.c

```diff
@@ -1740,12 +1740,18 @@ void clusterProcessGossipSection(clusterMsg *hdr, clusterLink *link) {
 /* IP -> string conversion. 'buf' is supposed to at least be 46 bytes.
  * If 'announced_ip' length is non-zero, it is used instead of extracting
  * the IP from the socket peer address. */
-void nodeIp2String(char *buf, clusterLink *link, char *announced_ip) {
+int nodeIp2String(char *buf, clusterLink *link, char *announced_ip) {
     if (announced_ip[0] != '\0') {
         memcpy(buf,announced_ip,NET_IP_STR_LEN);
         buf[NET_IP_STR_LEN-1] = '\0'; /* We are not sure the input is sane. */
+        return C_OK;
     } else {
-        connPeerToString(link->conn, buf, NET_IP_STR_LEN, NULL);
+        if (connPeerToString(link->conn, buf, NET_IP_STR_LEN, NULL) == C_ERR) {
+            serverLog(LL_WARNING, "ERROR get peer IP: %s", 
+                link->conn ? connGetLastError(link->conn) : "no link");
```

Comment:
I'm not sure this should be a warning, since it's being handled gracefully. It's an interesting failure, warranting a message, but it's not really an "error" which is what we use WARNING for.
```suggestion
            serverLog(LL_NOTICE, "Error converting peer IP to string: %s",
                link->conn ? connGetLastError(link->conn) : "no link");
```